CoursesManager.UI/ViewModels/Courses/CourseOverViewViewModel.cs

```diff
@@ -1,54 +1,108 @@
 ﻿using CoursesManager.MVVM.Commands;
 using CoursesManager.MVVM.Data;
 using CoursesManager.MVVM.Dialogs;
-using CoursesManager.UI.Utils;
+using CoursesManager.MVVM.Messages;
+using CoursesManager.MVVM.Navigation;
+using CoursesManager.UI.Messages;
 using CoursesManager.UI.Models;
 using CoursesManager.UI.Models.CoursesManager.UI.Models;
-using System;
-using System.Collections.Generic;
+using CoursesManager.UI.Models.Repositories.CourseRepository;
+using CoursesManager.UI.Utils;
+using CoursesManager.UI.ViewModels.Students;
 using System.Collections.ObjectModel;
-using System.Diagnostics;
-using System.Linq;
-using System.Reflection.Metadata;
-using System.Text;
-using System.Threading.Tasks;
 using System.Windows.Input;
 
 namespace CoursesManager.UI.ViewModels.Courses
 {
-    class CourseOverViewViewModel : ViewModel
+    internal class CourseOverViewViewModel : NavigatableViewModel
     {
-        public ICommand ChangeCourseCommand { get; set; }
+        private readonly ICourseRepository _courseRepository;
+        private readonly IDialogService _dialogService;
+        private readonly IMessageBroker _messageBroker;
 
+        public ICommand ChangeCourseCommand { get; set; }
         public ICommand DeleteCourseCommand { get; set; }
-        public Course CurrentCourse { get; set; }
+
+        private bool _isDialogOpen;
+        private Course? _currentCourse;
+
+        public bool IsDialogOpen
+        {
+            get => _isDialogOpen;
+            set => SetProperty(ref _isDialogOpen, value);
+        }
+
+        public Course CurrentCourse { get => _currentCourse!; set => SetProperty(ref _currentCourse, value); }
```

Comment:
Weet niet zeker of jij die uitroeptekens hebt toegevoegd maar weet je wat dat doet? En kan je garanderen dat dit klopt?